src/merkle/partial_mt/mod.rs

```diff
@@ -327,3 +403,36 @@ impl PartialMerkleTree {
         Ok(())
     }
 }
+
+// SERIALIZATION
+// ================================================================================================
+
+impl Serializable for PartialMerkleTree {
+    fn write_into<W: ByteWriter>(&self, target: &mut W) {
+        // write leaf nodes
+        target.write_u64(self.leaves.len() as u64);
+        for leaf_index in self.leaves.iter() {
+            leaf_index.write_into(target);
+            self.get_node(*leaf_index).expect("Leaf hash not found").write_into(target);
+        }
+    }
+}
+
+impl Deserializable for PartialMerkleTree {
+    fn read_from<R: ByteReader>(source: &mut R) -> Result<Self, DeserializationError> {
+        // add leaf nodes to the vector
+        let mut leaf_nodes = Vec::new();
+        let leaves_len = source.read_u64()?;
```

Comment:
Why not use `with_capacity()` when creating `leaf_nodes` vector? We know the number of leaves by this point.